Library/Homebrew/extend/os/linux/extend/pathname.rb

```diff
@@ -0,0 +1,21 @@
+class Pathname
+  # @private
+  def elf?
+    # List of file signatures: https://en.wikipedia.org/wiki/Executable_and_Linkable_Format#File_header
+    read(4) == "\x7fELF"
+  end
+
+  # @private
+  def dynamic_elf?
+    if !which("readelf").nil?
+      match = popen_read("readelf", "-l", to_path)[/ DYNAMIC /]
+      return false if match.empty?
+    elsif !which("file").nil?
+      match = `file -L -b #{self}` =~ /dynamic|shared/
+      return false if match.nil?
```

Comment:
I'd suggest…
```ruby
return !popen_read("file", "-L", "-b", to_path)[/dynamic|shared/].nil?
```
@MikeMcQuaid What's your preferred style in this case?

Reply:
done